File: core/src/com/oni/donjon/input/KeyboardInput.java
package com.oni.donjon.input;

import com.badlogic.gdx.Application;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input;
import com.badlogic.gdx.InputAdapter;
import com.oni.donjon.component.DirectionComponent;
import com.oni.donjon.data.GameData;

/**
 * @author Daniel Chesters (on 20/05/14).
 */
public class KeyboardInput extends InputAdapter {
    private GameData data;

    public void setData(GameData data) {
        this.data = data;
    }

    @Override public boolean keyDown(int keycode) {
        switch (keycode) {
            case Input.Keys.D:
            case Input.Keys.RIGHT:
                data.setPlayerDirection(DirectionComponent.Direction.RIGHT);
                break;
            case Input.Keys.Q:
            case Input.Keys.LEFT:
                data.setPlayerDirection(DirectionComponent.Direction.LEFT);
                break;
            case Input.Keys.Z:
            case Input.Keys.UP:
                data.setPlayerDirection(DirectionComponent.Direction.UP);
                break;
            case Input.Keys.S:
            case Input.Keys.DOWN:
                data.setPlayerDirection(DirectionComponent.Direction.DOWN);
                break;
            default:
                break;
        }
        if (Gdx.app.getLogLevel() == Application.LOG_DEBUG) {
            debugMessage(keycode);
        }
        return true;
    }

    @Override public boolean keyUp(int keycode) {
        switch (keycode) {
            case Input.Keys.D:
            case Input.Keys.RIGHT:
            case Input.Keys.Q:
            case Input.Keys.LEFT:
            case Input.Keys.Z:
            case Input.Keys.UP:
            case Input.Keys.S:
            case Input.Keys.DOWN:
                data.setPlayerDirection(DirectionComponent.Direction.NONE);
                break;
            default:
                break;
        }

        return true;
    }

    private void debugMessage(int keycode) {
        Gdx.app.debug("Keyboard", Input.Keys.toString(keycode));
    }
}

File: core/src/main/kotlin/com/oni/donjon/input/KeyboardInput.kt
package com.oni.donjon.input

import com.badlogic.gdx.Input
import com.oni.donjon.component.DirectionComponent
import com.oni.donjon.data.GameData
import ktx.app.KtxInputAdapter
import ktx.log.logger

/**
 * @author Daniel Chesters (on 20/05/14).
 */
class KeyboardInput : KtxInputAdapter {
    companion object {
        val log = logger<KeyboardInput>()
    }

    override fun keyDown(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.D, Input.Keys.RIGHT -> GameData.setPlayerDirection(DirectionComponent.Direction.RIGHT)
            Input.Keys.Q, Input.Keys.LEFT -> GameData.setPlayerDirection(DirectionComponent.Direction.LEFT)
            Input.Keys.Z, Input.Keys.UP -> GameData.setPlayerDirection(DirectionComponent.Direction.UP)
            Input.Keys.S, Input.Keys.DOWN -> GameData.setPlayerDirection(DirectionComponent.Direction.DOWN)
        }
        debugMessage(keycode)
        return true
    }

    override fun keyUp(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.D, Input.Keys.RIGHT, Input.Keys.Q, Input.Keys.LEFT,
            Input.Keys.Z, Input.Keys.UP, Input.Keys.S, Input.Keys.DOWN ->
                GameData.setPlayerDirection(DirectionComponent.Direction.NONE)
        }

        return true
    }

    private fun debugMessage(keycode: Int) {
        log.debug { Input.Keys.toString(keycode) }
    }
}
